### PR TITLE
feat: add "display_order" column to coder_parameter to keep parameters sorted in UI

### LINKED ISSUE #8193: coder_parameter: add "order" DB column to ensure file ordering
With `synchronize_seqscans = on` (default?), rich parameters are fetched from the database table `template_version_parameters` in the right order, as they exist in the .tf file. 

Unfortunately, it is non-deterministic with `synchronize_seqscans = off`, so we need to introduce an extra db column to ensure the correct ordering.

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/8193

This PR implements the "display_order" db column (`ORDER` is a reserved word), which is used to ensure the order of parameters based on the `coder_parameter` TF [property](https://github.com/coder/terraform-provider-coder/pull/134).

### CODE REVIEW

File: coderd/database/migrations/000132_parameters_order.up.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE template_version_parameters ADD COLUMN display_order integer NOT NULL DEFAULT 0;
+
+COMMENT ON COLUMN template_version_parameters.display_order
+IS 'Display order';
```
Comment (mafredri) [inline coderd/database/migrations/000132_parameters_order.up.sql:4]:
```suggestion
IS 'Specifies the order in which to display parameters in user interfaces.';
```

File: coderd/database/queries/templateversionparameters.sql [hunk 2]
```diff
@@ -35,8 +36,9 @@ VALUES
...
         $14,
         $15,
-		$16
+        $16,
+        $17
     ) RETURNING *;
 
 -- name: GetTemplateVersionParameters :many
-SELECT * FROM template_version_parameters WHERE template_version_id = $1;
+SELECT * FROM template_version_parameters WHERE template_version_id = $1 ORDER BY display_order ASC, LOWER(name) ASC;
```
Comment (mafredri) [inline coderd/database/queries/templateversionparameters.sql:44]: Thanks for using explicit `ASC`, helps to convey the intent. ☺️

**Review by mafredri — APPROVED** [review]
This turned out nicely, nice to see a bunch of code removed! 😍

File: coderd/database/migrations/000132_parameters_order.up.sql (hunk 1 above)
Comment (mtojek) [reply coderd/database/migrations/000132_parameters_order.up.sql]: Fixed, thanks for suggestion

**Pull request merged**